lib/SILOptimizer/FunctionSignatureTransforms/ExistentialTransform.cpp

```diff
@@ -0,0 +1,547 @@
+      case ExistentialRepresentation::Opaque: {
+        /// Create this sequence for init_existential_addr.:
+        /// bb0(%0 : $*T):
+        /// %3 = alloc_stack $P
+        /// %4 = init_existential_addr %3 : $*P, $T
+        /// copy_addr %0 to [initialization] %4 : $*T
+        /// destroy_addr %0 : $*T
+        /// %7 = open_existential_addr immutable_access %3 : $*P to
+        /// $*@opened("105977B0-D8EB-11E4-AC00-3C0754644993") P
+        auto *ASI =
+            NewFBuilder.createAllocStack(InsertLoc, ArgDesc.Arg->getType());
+        Arg2AllocStackMap.insert(
+            std::pair<int, SILInstruction *>(ArgDesc.Index, ASI));
+
+        auto *EAI = NewFBuilder.createInitExistentialAddr(
+            InsertLoc, ASI, NewArg->getType().getASTType()->getCanonicalType(),
+            NewArg->getType(), Conformances);
+        /// If DestroyAddr is already there, then do not use [take].
+        NewFBuilder.createCopyAddr(
+            InsertLoc, NewArg, EAI,
+            ExistentialArgDescriptor[ArgDesc.Index].DestroyAddrUse
+                ? IsTake_t::IsNotTake
+                : IsTake_t::IsTake,
+            IsInitialization_t::IsInitialization);
+        if (ExistentialArgDescriptor[ArgDesc.Index].DestroyAddrUse) {
+          NewFBuilder.createDestroyAddr(InsertLoc, NewArg);
+        } else {
+          MissingDestroyUse = true;
+        }
+        BranchArgs.push_back(ASI);
+        break;
+      }
+      case ExistentialRepresentation::Class: {
+        ///  Create an init_existential.
+        /// %5 = init_existential_ref %0 : $T : $T, $P
+        auto *InitRef = NewFBuilder.createInitExistentialRef(
+            InsertLoc, ArgDesc.Arg->getType(),
+            NewArg->getType().getASTType()->getCanonicalType(), NewArg,
+            Conformances);
+        BranchArgs.push_back(InitRef);
+        break;
+      }
+      default: {
+        llvm_unreachable("Unhandled existential type in ExistentialTransform!");
+        break;
+      }
+      };
+    } else {
+      auto Ty = params[ArgDesc.Index].getType();
+      if (!(Ty->hasTypeParameter() || Ty->hasArchetype())) {
+        auto NewArg = NewFBody->createFunctionArgument(ArgDesc.Arg->getType(),
+                                                       ArgDesc.Decl);
+        NewArg->setOwnershipKind(ValueOwnershipKind(
+            M, ArgDesc.Arg->getType(), ArgDesc.Arg->getArgumentConvention()));
+        BranchArgs.push_back(NewArg);
+      } else {
+        SILType GenericSILType =
+            M.Types.getLoweredType(NewF->mapTypeIntoContext(Ty));
+        auto *NewArg = NewFBody->createFunctionArgument(GenericSILType);
+        NewArg->setOwnershipKind(ValueOwnershipKind(
+            M, GenericSILType, ArgDesc.Arg->getArgumentConvention()));
+        BranchArgs.push_back(NewArg);
+      }
+    }
+  }
+
+  /// Then we splice the body of F to NewF after the first basic block.
+  auto It = NewF->begin();
+  It++;
+  NewF->getBlocks().splice(It, F->getBlocks());
```

Comment:
Instead of  splice: use a TypeSubstCloner with a Builder and SubstitutionMap and then call the API visitSILFunction(original function F);